karabo_data/geometry2.py

```diff
@@ -205,6 +207,40 @@ def inspect(self, frontview=True):
 
         return ax
 
+    @classmethod
+    def from_crystfel_geom(cls, filename):
+        """Read a CrystFEL format (.geom) geometry file.
+
+        Returns a new geometry object.
+        """
+        geom_dict = load_crystfel_geometry(filename)
+        modules = []
+        for p in range(cls.n_modules):
+            tiles = []
+            modules.append(tiles)
+            for a in range(cls.n_asics_per_module):
+                d = geom_dict['panels']['p{}a{}'.format(p, a)]
+                tiles.append(GeometryFragment.from_panel_dict(d))
+        return cls(modules, filename=filename)
+
+    def write_crystfel_geom(self, filename):
+        """Write this geometry to a CrystFEL format (.geom) geometry file."""
+        from . import __version__
+
+        panel_chunks = []
+        for p, module in enumerate(self.modules):
+            for a, fragment in enumerate(module):
+                panel_chunks.append(fragment.to_crystfel_geom(p, a))
```

Comment:
The creation of the chunk for an individual panel is a bit AGIPD-specific at the moment - in particular around the min/max slow-scan and fast-scan coordinates defining where the panel is.

Reply:
Yes! I added some additional keywords to the fragment header. I've tested it with all three available detectors. It seems to work quite fine.

I am inclined to say that the CFEL format should become our default format and that we should push users using this format (or at least encourage them). I think this PR would be a great step into this direction.